Xwt.XamMac/Xwt.Mac/WindowBackend.cs

```diff
@@ -211,33 +210,45 @@ void IBackend.EnableEvent (object eventId)
 			if (eventId is WindowFrameEvent) {
 				var @event = (WindowFrameEvent)eventId;
 				switch (@event) {
-				case WindowFrameEvent.BoundsChanged:
-					DidResize += HandleDidResize;
-					DidMove += HandleDidResize;
-					break;
-				case WindowFrameEvent.Hidden:
-					EnableVisibilityEvent (@event);
-					this.WillClose += OnWillClose;
-					break;
-				case WindowFrameEvent.Shown:
-					EnableVisibilityEvent (@event);
-					break;
-				case WindowFrameEvent.CloseRequested:
-					WindowShouldClose = OnShouldClose;
-					break;
+					case WindowFrameEvent.Hidden:
+					case WindowFrameEvent.Shown:
+						if (!VisibilityEventsEnabled())
+						{
+							ContentView.AddObserver(this, HiddenProperty, NSKeyValueObservingOptions.New, IntPtr.Zero);
```

Comment:
Store the observer and dispose it on window dispose.

Reply:
hhm, `AddObserver` is `void` (it's not like NotificationManager) and we remove it here: https://github.com/mono/xwt/blob/512728f98e8390de4919b85ee7177a8e1b35994b/Xwt.XamMac/Xwt.Mac/WindowBackend.cs#L477

Reply:
Ah, I see. Good.